pkg/operator/controller/ingress/load_balancer_service.go

```diff
@@ -812,3 +908,180 @@ func loadBalancerSourceRangesMatch(ic *operatorv1.IngressController, current *co
+	if a, ok := service.Annotations[awsLBSubnetsAnnotation]; ok {
+		var subnets []string
+		a = strings.TrimSpace(a)
+		if len(a) > 0 {
+			subnets = strings.Split(a, ",")
+		}
+
+		// Cast the slice of strings to AWSSubnets object while distinguishing between subnet IDs and Names.
+		for _, subnet := range subnets {
+			if strings.HasPrefix(subnet, "subnet-") {
+				awsSubnets.IDs = append(awsSubnets.IDs, operatorv1.AWSSubnetID(subnet))
+			} else {
+				awsSubnets.Names = append(awsSubnets.Names, operatorv1.AWSSubnetName(subnet))
+			}
+		}
+	}
+
+	// Return nil if no subnets are found.
+	if len(awsSubnets.IDs) == 0 && len(awsSubnets.Names) == 0 {
+		return nil
+	}
+
+	return awsSubnets
+}
+
+// compareAWSSubnets compares two AWSSubnets objects and returns a boolean
+// whether they are equal are not.
+func compareAWSSubnets(subnets1, subnets2 *operatorv1.AWSSubnets) bool {
+	// If they are both nil, they are equal.
+	if subnets1 == nil && subnets2 == nil {
+		return true
+	}
+
+	// If one is nil and the other is not, they are equal only if the non-nil one is empty.
+	if subnets1 == nil {
+		return len(subnets2.IDs) == 0 && len(subnets2.Names) == 0
+	}
+	if subnets2 == nil {
+		return len(subnets1.IDs) == 0 && len(subnets1.Names) == 0
+	}
+
+	// If they both are non-nil, compare the length first, then do a more detailed comparison if needed.
+	if len(subnets1.Names) != len(subnets2.Names) || len(subnets1.IDs) != len(subnets2.IDs) {
+		return false
+	}
+
+	// Create maps to track the IDs from each subnet object for comparison.
+	subnetIDMap1 := make(map[operatorv1.AWSSubnetID]bool)
+	subnetIDMap2 := make(map[operatorv1.AWSSubnetID]bool)
+	for _, id := range subnets1.IDs {
+		subnetIDMap1[id] = true
+	}
+	for _, id := range subnets2.IDs {
+		subnetIDMap2[id] = true
+	}
+	// Check if maps contain the same IDs.
+	for id := range subnetIDMap1 {
+		if !subnetIDMap2[id] {
+			return false
+		}
+	}
+	// Create maps to track the names from each subnet object for comparison.
+	subnetNameMap1 := make(map[operatorv1.AWSSubnetName]bool)
+	subnetNameMap2 := make(map[operatorv1.AWSSubnetName]bool)
+	for _, name := range subnets1.Names {
+		subnetNameMap1[name] = true
+	}
+	for _, name := range subnets2.Names {
+		subnetNameMap2[name] = true
+	}
+	// Check if maps contain the same names.
+	for name := range subnetNameMap1 {
+		if !subnetNameMap2[name] {
+			return false
+		}
+	}
+
+	return true
+}
+
+// awsSubnetsExist checks if any subnets exist in an AWSSubnets structure.
+func awsSubnetsExist(subnets *operatorv1.AWSSubnets) bool {
+	return subnets != nil && (len(subnets.Names) > 0 || len(subnets.IDs) > 0)
+}
+
+// JoinAWSSubnets joins an AWS Subnets object into a string seperated by sep.
+func JoinAWSSubnets(subnets *operatorv1.AWSSubnets, sep string) string {
+	if subnets == nil {
+		return ""
+	}
+	joinedSubnets := ""
+	subnetCount := 0
+	for _, subnet := range subnets.IDs {
+		if subnetCount > 0 {
+			joinedSubnets += sep
```

Comment:
Can't we use `strings.Join` function ?  
joinedSubnetID = strings.Join(subnets.IDs, sep)
joinedSubnetNames = strings.Join(subnets.Names, sep)
joinedSubnets = joinedSubnetID + joinedSubnetNames